src/compas_fab/robots/robot.py

```diff
@@ -1076,30 +996,33 @@ def inverse_kinematics(self, frame_WCF, start_configuration=None,
                 attached_collision_meshes = [self.attached_tool.attached_collision_mesh]
 
         # The returned joint names might be more than the requested ones if there are passive joints present
-        joint_positions, joint_names = self.client.inverse_kinematics(frame_RCF, base_link,
-                                                                      group, joint_names, joint_positions,
+        joint_positions, joint_names = self.client.inverse_kinematics(self,
+                                                                      frame_WCF_scaled,
+                                                                      group, start_configuration_scaled,
                                                                       avoid_collisions, constraints, attempts,
                                                                       attached_collision_meshes)
-        joint_types = [self.get_joint_by_name(n).type for n in joint_names]
-
-        joint_positions = self._scale_joint_values(joint_positions, joint_names, self.scale_factor)
-
-        # build configuration including passive joints
-        configuration = Configuration(joint_positions, joint_types)
+        if full_joint_state:
+            # build configuration including passive joints, but no sorting
+            joint_types = self.get_joint_types_by_names(joint_names)
+            configuration = Configuration(joint_positions, joint_types, joint_names)
+        else:
+            # sort values for group configuration
+            joint_state = dict(zip(joint_names, joint_positions))
+            group_joint_names = self.get_configurable_joint_names(group)
+            values = [joint_state[name] for name in group_joint_names]
+            configuration = Configuration(values, self.get_configurable_joint_types(group), group_joint_names)
 
-        # return only group configuration
-        # NOTE: it might actually make more sense to return
-        # the configuration instance without extracting the group's config
-        # because we lose the passive joint info
-        return self.get_group_configuration(group, configuration)
+        return configuration.scaled(self.scale_factor)
 
-    def forward_kinematics(self, configuration, group=None, backend=None, link_name=None):
+    def forward_kinematics(self, full_configuration, group=None, backend=None, link_name=None):
```

Comment:
Here I'm unsure. The renaming to `full_configuration` is somewhat unneeded because it actually still takes a non-full config, and it will fill it up with zeros, right?

Reply:
yes, correct, we can change it to `configuration`